test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/DrainModeResumeEndToEndTests.cs

```diff
@@ -0,0 +1,113 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.EndToEnd
+{
+    [Trait(TestTraits.Category, TestTraits.EndToEnd)]
+    [Trait(TestTraits.Group, TestTraits.DrainModeEndToEnd)]
+    public class DrainModeResumeEndToEndTests : ResumeTestFixture
+    {
+        [Fact]
+        public async Task DrainModeEnabled_RunningHost_StartsNewHost_ReturnsOk()
+        {
+            // Validate the drain state is "Disabled" initially
+            var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+            var responseString = response.Content.ReadAsStringAsync().Result;
+            var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+            Assert.Equal(status.State, DrainModeState.Disabled);
+
+            ManualResetEvent resetEvent = new ManualResetEvent(false);
+            _ = Task.Run(async () =>
+            {
+                // Put the host in drain mode
+                response = await SamplesTestHelpers.InvokeDrain(this);
+
+                // Validate the drain state is changed to "InProgress"
+                await TestHelpers.Await(async () =>
+                {
+                    var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+                    var responseString = response.Content.ReadAsStringAsync().Result;
+                    var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+                    return status.State == DrainModeState.InProgress;
+                }, 20000);
+
+                // Validate host is "Running" after resume is called
+                await TestHelpers.Await(async () =>
+                {
+                    var response = await SamplesTestHelpers.InvokeResume(this);
+                    var responseString = response.Content.ReadAsStringAsync().Result;
+                    var status = JObject.Parse(responseString);
+
+                    return (string)status["hostStatus"] == ScriptHostState.Running.ToString();
+                }, 20000);
+
+                // Validate the drain state is changed to "Disabled"
+                await TestHelpers.Await(async () =>
+                {
+                    var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+                    var responseString = response.Content.ReadAsStringAsync().Result;
+                    var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+                    return status.State == DrainModeState.Disabled;
+                }, 20000);
+
+                resetEvent.Set();
+            });
+
+            response = await SamplesTestHelpers.InvokeHttpTrigger(this, "HttpTrigger");
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            bool result = resetEvent.WaitOne(30000);
+            Assert.True(result);
```

Comment:
Might be good to call host/status before and after to compare InstanceId which will be different, verifying a new host is running?

Reply:
Good idea!

Reply:
So the InstanceId == "" when I make the `/status` request in the E2E test. Any suggestions?

Reply:
It's the latter instance ID I was talking about, which I incorrectly assumed was what was returned via HostStatus.InstanceId. The goal is for the test to ensure that a new host instance is running. I think you can do that using the [HostInstanceId](https://github.com/Azure/azure-functions-host/blob/cbf9ad1b107c52531975f622d7f50f63ab31e592/test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/EndToEndTestFixture.cs#L75) property in the base test fixture.

Reply:
I'll give that a shot, thanks for the pointer :)